packages/venia-ui/lib/components/CheckoutPage/PaymentInformation/creditCard.js

```diff
@@ -124,20 +124,17 @@ const CreditCard = props => {
      * We write this function in `venia-ui` and not in the `peregrine` talon
      * because it references `isRequired` which is a `venia-ui` function.
      */
-    const isFieldRequired = useCallback(
-        value => {
-            if (isBillingAddressSame) {
-                /**
-                 * Informed validator functions return `undefined` is
-                 * validation is `true`
-                 */
-                return undefined;
-            } else {
-                return isRequired(value);
-            }
-        },
-        [isBillingAddressSame]
-    );
+    const isFieldRequired = useCallback((value, { isBillingAddressSame }) => {
```

Comment:
What's up with this change?

Reply:
Something changed with the way informed caches prop function given as `validate` prop. It was saving the function which has `isBillingAddressSame` as a closure parameter. Due to this, in subsequent renders, even if `isBillingAddressSame` was changing, the value inside of that function wasn't. Has something to do with how `informed` is handling its props. But the good thing is `informed` provides a second argument to the validate function which is an object of all the form values. Hence I changed the code to pick the required value from a different source.